src/components/VImg/VImg.ts

```diff
@@ -0,0 +1,197 @@
+import '../../stylus/components/_images.styl'
+
+// Types
+import { VNode } from 'vue'
+import { PropValidator } from 'vue/types/options'
+
+// Components
+import VResponsive from '../VResponsive'
+
+// Utils
+import { consoleError } from '../../util/console'
+
+// not intended for public use, this is passed in by vuetify-loader
+export interface srcObject {
+  src: string
+  srcset?: string
+  lazySrc: string
+  aspect: number
+}
+
+export default VResponsive.extend({
+  name: 'v-img',
+
+  props: {
+    alt: String,
+    contain: Boolean,
+    src: {
+      type: [String, Object],
+      required: true
+    } as PropValidator<string | srcObject>,
+    lazySrc: String,
+    srcset: String,
+    sizes: String,
+    position: {
+      type: String,
+      default: 'center center'
+    },
+    transition: {
+      type: String,
+      default: 'fade-transition'
+    }
+  },
+
+  data () {
+    return {
+      currentSrc: '', // Set from srcset
+      image: null as HTMLImageElement | null,
+      isLoading: true,
+      calculatedAspectRatio: undefined as number | undefined
+    }
+  },
+
+  computed: {
+    computedSrc (): string {
+      return typeof this.src === 'string' ? this.src : this.src.src
+    },
+    computedSrcset (): string | undefined {
+      return typeof this.src === 'string' ? this.srcset : this.src.srcset
+    },
+    computedLazySrc (): string {
+      return typeof this.src === 'string'
```

Comment:
With the above suggestion this could be simplified

```
return (this.srcIsString && this.lazySrc) || this.src.lazySrc
```

Also it might not be a great assumption that if the left side of the ternary is false, that `this.src` is an object containing a **lazySrc** property.

Reply:
`src` is required, and vuetify-loader will always supply lazySrc. Passing in an object is not public API.